plugins/module_utils/waiters.py

```diff
@@ -162,6 +162,19 @@
                 },
             ]
         },
+        "SubnetAvailable": {
```

Comment:
There seems a boto3 built-in waiter has already been implemented for doing this https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2.html#EC2.Waiter.VpcAvailable.
Does it makes sense to you?

Reply:
The trouble with the default boto3 waiter is that if it hits the API rate limiting errors it bombs straight out.  This custom waiter treats the rate limiting error as temporary and just retries after the next delay.

Reply:
@tremble Thank you for explaining. Hence, it makes sense using the custom one.